config/policy.go

```diff
@@ -0,0 +1,254 @@
+	bytes, err := json.Marshal(o)
+
+	if err != nil {
+		panic(fmt.Sprintf("failed to convert to json with error: %s", err.Error()))
+	}
+
+	return string(bytes)
+}
+
+// WithPolicyConfig configures this edge with the provided policy configuration
+// passed as a json string and overwrites any prevously-set traffic policy
+// https://ngrok.com/docs/http/traffic-policy
+func WithPolicyConfig(jsonStr string) interface {
+	HTTPEndpointOption
+	TLSEndpointOption
+	TCPEndpointOption
+	JsonConvertible
+} {
+	p := &policy{}
+	if err := json.Unmarshal([]byte(jsonStr), p); err != nil {
+		panic("invalid json for policy configuration")
+	}
+
+	return p
+}
+
+// WithPolicy configures this edge with the given traffic and overwrites any
+// previously-set traffic policy
+// https://ngrok.com/docs/http/traffic-policy/
+func WithPolicy(opts ...policyOption) interface {
+	HTTPEndpointOption
+	TLSEndpointOption
+	TCPEndpointOption
+	JsonConvertible
+} {
+	cfg := &policy{}
+	applyOpts(cfg, opts...)
+
+	return cfg
+}
+
+// WithInboundRules adds the provided policy rules to the inbound
+// set of the given policy.
+// The order in which policies are specified is observed.
+func WithInboundRules(opts ...policyRuleSetOption) policyOption {
+	rules := []policyRule{}
+	applyOpts(&rules, opts...)
+
+	return inboundRules(rules)
+}
+
+// WithOutboundRules adds the provided policy to be outbound
+// set of the given policy.
+// The order in which policies are specified is observed.
+func WithOutboundRules(opts ...policyRuleSetOption) policyOption {
+	rules := []policyRule{}
+	applyOpts(&rules, opts...)
+
+	return outboundRules(rules)
+}
+
+// WithPolicyRule provides a policy rule built from the given options.
+func WithPolicyRule(opts ...policyRuleOption) interface {
+	policyRuleSetOption
+	JsonConvertible
+} {
+	pr := policyRule{}
+	applyOpts(&pr, opts...)
+
+	return pr
+}
+
+// WithPolicyName sets the provided name on a policy rule.
+func WithPolicyName(n string) policyRuleOption {
+	return optionFunc[*policyRule](
+		func(r *policyRule) {
+			r.Name = n
+		})
+}
+
+// WithPolicyExpression appends the provided CEL statement to a policy rule's expressions.
+func WithPolicyExpression(e string) policyRuleOption {
+	return optionFunc[*policyRule](
+		func(r *policyRule) {
+			r.Expressions = append(r.Expressions, e)
+		})
+}
+
+// WithPolicyAction appends the provided action to the set of the policy rule.
+// The order the actions are specified is observed.
+func WithPolicyAction(opts ...policyActionOption) interface {
+	policyRuleOption
+	JsonConvertible
+} {
+	a := action{}
+	applyOpts(&a, opts...)
+
+	return a
+}
+
+// WithActionType sets the provided type for this action. Type must be specified.
+func WithPolicyActionType(t string) policyActionOption {
+	return optionFunc[*action](func(a *action) { a.Type = t })
+}
+
+// WithConfig sets the provided map as the configuration for this action
+func WithPolicyActionConfig(c map[string]any) policyActionOption {
```

Comment:
Our SDK design says to accept a json string here, but is that really what we want? Or should we accept a map like this?